### ISSUE TITLE
完善AOT编译链路

### ISSUE THREAD

**Issue by yfchen-byted** [BD-Seed-HHW/xpu_graph#308]
### Describe the feature🚀

XPUGraph的现有工作是基于JIT链路实现的。
考虑到
1 传统的JIT链路会造成复杂的guard检查，运行时开销大且难以干涉
2 一些场景模型结构比较稳定，除了batch以外变化不大
通过AOT链路能够减少运行时overhead

### References

Torch文档：[torch.export](https://pytorch.org/docs/main/export.html)
推荐阅读：[Ways to use torch export](https://blog.ezyang.com/2024/12/ways-to-use-torch-export/)

### Environment details

PyTorch: >= 2.6

**Comment by yfchen-byted** [comment]
以NPU上目前的draft为例； #188 由于序列化方面存在的问题，目前所有的引入新kernel的pass都无法应用，所以目前加了开关跳过了所有的优化pass，且只应用aot_inductor作为codegen。不过从其他硬件的cache经验看，这个问题应该是可以解决的

**Comment by wwens7** [comment]
这里“复杂的guard检查”可以用 skip_guard_eval_unsafe 吗？

**Comment by LeoLau94** [comment]
> 看了推荐阅读，这里“复杂的guard检查”可以用博客里的 skip_guard_eval_unsafe 吗？

博客里有提到这个么？

**Comment by yfchen-byted** [comment]
> 这个方案看起来没有支持vendor compiler

这个方案是指哪一个实现？

**Comment by yfchen-byted** [comment]
> 后端有inductor的实现/注册

需要，这是完全为了AOT Inductor而实现的链路

**Comment by yfchen-byted** [comment]
走不走cudagraph其实跟是否是C/CPP wrapper是正交的

**Comment by LeoLau94** [comment]
> > 后端有inductor的实现/注册
> 
> 需要，这是完全为了AOT Inductor而实现的链路

aot inductor会用到一些融合操作吗？还是直接做简单的cpp接口剥离

**Comment by yfchen-byted** [comment]
> > > 后端有inductor的实现/注册
> > 
> > 
> > 需要，这是完全为了AOT Inductor而实现的链路
> 
> aot inductor会用到一些融合操作吗？还是直接做简单的cpp接口剥离

相当于普通的inductor + CPP接口

**Comment by LeoLau94** [comment]
> 走不走cudagraph其实跟是否是C/CPP wrapper是正交的，这是两个问题；从MLU经验看，CPP wrapper也可以用cudagraphs

这个如果想要直接序列化成带cuda graph的调用，是需要在inductor里面去调用cuda graph吧。

作者是知道gpu inductor里没调用，所以说没支持？然后也认为可以不支持或者在外部使用with graph去capture？

**Comment by LeoLau94** [comment]
PyTorch 2.5.0 可以设置这里的config
/usr/local/lib/python3.10/site-packages/torch/_dynamo/config.py
